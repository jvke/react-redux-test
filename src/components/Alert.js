import React from 'react';
import styled from 'styled-components';
import { darken, lighten, transparentize } from 'polished';

const Alert = ({ children, ...otherProps }) => (
  <p {...otherProps}>{children}</p>
);

const StyledAlert = styled(Alert)`
  border: solid 3px ${props => lighten(0.05, props.theme.brand.light)};
  padding: 10px 20px;
  background: ${props => transparentize(0.3, darken(0.1, props.theme.brand.light))};
  color: ${props => props.theme.textColor};
  border-radius: 4px;
`;


const ErrorAlert = StyledAlert.extend`
  border: solid 3px ${props => darken(0.1, props.theme.alert.error)};
  background: ${props => transparentize(0.3, darken(0.3, props.theme.alert.error))}
`;

export default StyledAlert;
export { ErrorAlert };
