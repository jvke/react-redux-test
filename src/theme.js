export default {
  brand: {
    light: 'rgb(8, 62, 101)',
    dark: 'rgb(1, 20, 46)',
  },
  alert: {
    error: '#FF0000'
  },
  textColor: '#EEEEEE'
}
