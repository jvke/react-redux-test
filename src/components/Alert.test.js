import React from 'react';
import ReactDOM from 'react-dom';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import Alert, { ErrorAlert } from './Alert';
import theme from '../theme';


describe('Alert', () => {
  it('renders with no props specified', () => {
    const tree = renderer
      .create(<Alert theme={theme} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with props specified', () => {
    const tree = renderer
      .create(<Alert theme={theme} a={1} b={2} c={3} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders error alerts correctly', () => {
    const tree = renderer
      .create(<ErrorAlert theme={theme} a={1} b={2} c={3} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})
