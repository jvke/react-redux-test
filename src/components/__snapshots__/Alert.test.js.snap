// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders error alerts correctly 1`] = `
.c0 {
  border: solid 3px #0a4d7d;
  padding: 10px 20px;
  background: rgba(4,33,54,0.7);
  color: #EEEEEE;
  border-radius: 4px;
  border: solid 3px #c00;
  background: rgba(102,0,0,0.7);
}

<p
  a={1}
  b={2}
  c={3}
  className="c0"
  theme={
    Object {
      "alert": Object {
        "error": "#FF0000",
      },
      "brand": Object {
        "dark": "rgb(1, 20, 46)",
        "light": "rgb(8, 62, 101)",
      },
      "textColor": "#EEEEEE",
    }
  }
/>
`;

exports[`Alert renders with no props specified 1`] = `
.c0 {
  border: solid 3px #0a4d7d;
  padding: 10px 20px;
  background: rgba(4,33,54,0.7);
  color: #EEEEEE;
  border-radius: 4px;
}

<p
  className="c0"
  theme={
    Object {
      "alert": Object {
        "error": "#FF0000",
      },
      "brand": Object {
        "dark": "rgb(1, 20, 46)",
        "light": "rgb(8, 62, 101)",
      },
      "textColor": "#EEEEEE",
    }
  }
/>
`;

exports[`Alert renders with props specified 1`] = `
.c0 {
  border: solid 3px #0a4d7d;
  padding: 10px 20px;
  background: rgba(4,33,54,0.7);
  color: #EEEEEE;
  border-radius: 4px;
}

<p
  a={1}
  b={2}
  c={3}
  className="c0"
  theme={
    Object {
      "alert": Object {
        "error": "#FF0000",
      },
      "brand": Object {
        "dark": "rgb(1, 20, 46)",
        "light": "rgb(8, 62, 101)",
      },
      "textColor": "#EEEEEE",
    }
  }
/>
`;
